src/k4a_ros_device.cpp

```diff
@@ -300,6 +302,11 @@ K4AROSDevice::~K4AROSDevice()
   // Start tearing down the publisher threads
   running_ = false;
 
+  // Join the publisher thread
+  RCLCPP_INFO(this->get_logger(),"Joining diagnostics thread");
+  update_diagnostics_thread_.join();
+  RCLCPP_INFO(this->get_logger(),"Diagnostics thread joined");
+
```

Comment:
joining as I know means to wait for thread job to finish, but here job is endless, killing thread or stopping is what I would expect here

Reply:
before they call join on the threads, they set _running to false to stop the loop in the threads

Reply:
I could try to run the diagnostics without the use of the thread

Reply:
nah let's keep the thread - just check that it will shut down. Maybe we need another flag then, to shut it down?